Sources/Classes/RSDBPersistentManager.m

```diff
@@ -70,77 +79,17 @@ - (void)createDB:(RSDBEncryption * __nullable)dbEncryption {
-                // no key is provided
-                // delete encrypted database; then open unencrypted database
-                // all previous events will be deleted
-                [RSLogger logError:@"RSDBPersistentManager: createDB: No key is provided. Deleting encrypted DB and creating a new unencrypted DB"];
-                [RSUtils removeFile:ENCRYPTED_DB_NAME];
-                [self openUnencryptedDB];
-            } else {
-                int code = [self openEncryptedDB:dbEncryption.key];
-                switch (code) {
-                        // when key is correct
-                        // decyprt database; then open unencrypted database
-                    case SQLITE_OK: {
-                        code = [self decryptDB:dbEncryption.key];
-                        if (code == SQLITE_OK) {
-                            [self closeDB];
-                            [RSUtils removeFile:ENCRYPTED_DB_NAME];
-                            [self openUnencryptedDB];
-                        } else {
-                            [RSLogger logError:[NSString stringWithFormat:@"RSDBPersistentManager: createDB: Failed to decrypt, error code: %d", code]];
-                        }
-                    }
-                        break;
-                        // when key is wrong
-                        // delete encrypted database; then open unencrypted database
-                        // all previous events will be deleted
-                    case SQLITE_NOTADB: {
-                        [RSLogger logError:@"RSDBPersistentManager: createDB: Wrong key is provided. Deleting encrypted DB and creating a new unencrypted DB"];
-                        [self closeDB];
-                        [RSUtils removeFile:ENCRYPTED_DB_NAME];
-                        [self openUnencryptedDB];
-                    }
-                        break;
-                    default:
-                        [RSLogger logError:[NSString stringWithFormat:@"RSDBPersistentManager: createDB: Failed to decrypt, error code: %d", code]];
-                        break;
-                }
-            }
-        }
-    } else {
-        if (isEncryptionNeeded) {
-            // encyprt database; then open encrypted database
-            [self openUnencryptedDB];
-            int code = [self encryptDB:dbEncryption.key];
-            if (code == SQLITE_OK) {
-                [self closeDB];
-                [RSUtils removeFile:UNENCRYPTED_DB_NAME];
-                [self openEncryptedDB:dbEncryption.key];
-            } else {
-                [RSLogger logError:[NSString stringWithFormat:@"RSDBPersistentManager: createDB: Failed to encrypt, error code: %d", code]];
-            }
-        } else {
-            // open unencrypted database
-            [self openUnencryptedDB];
-        }
+        return;
     }
-    [self->lock unlock];
+
+    // when only encrypted db exists
+    if (isEncryptedDBExists) {
```

Comment:
You can use `else if` for better understanding as this has not very much complexity.

Reply:
Sure

Reply:
Addressed this